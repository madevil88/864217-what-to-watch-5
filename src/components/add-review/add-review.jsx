import React from "react";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute} from "../../const";
import {getFilms} from "../../store/selectors";
import MainProps from "../main/main-props";
import AddReviewForm from "../add-review-form/add-review-form";
import withInputValue from "../../hocs/with-input-value/with-input-value";
import UserBlock from "../user-block/user-block";

const AddReviewFormWrapped = withInputValue(AddReviewForm);

const AddReview = (props) => {
  const {films, id} = props;
  const currentFilm = films.allFilms[id - 1];

  return (
    <section className="movie-card movie-card--full">
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src="/img/bg-the-grand-budapest-hotel.jpg" alt="The Grand Budapest Hotel" />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header">
          <div className="logo">
            <Link to={AppRoute.ROOT} className="logo__link">
              <span className="logo__letter logo__letter--1">W</span>
              <span className="logo__letter logo__letter--2">T</span>
              <span className="logo__letter logo__letter--3">W</span>
            </Link>
          </div>

          <nav className="breadcrumbs">
            <ul className="breadcrumbs__list">
              <li className="breadcrumbs__item">
                <Link to={`/films/${currentFilm.id}`} className="breadcrumbs__link">{currentFilm.name}</Link>
              </li>
              <li className="breadcrumbs__item">
                <a className="breadcrumbs__link">Add review</a>
              </li>
            </ul>
          </nav>

          <UserBlock />
        </header>

        <div className="movie-card__poster movie-card__poster--small">
          <img src={currentFilm.poster_image} alt={currentFilm.name} width="218" height="327" />
        </div>
      </div>
      <AddReviewFormWrapped
        id={id}
      />
    </section>
  );
};

AddReview.propTypes = MainProps.propTypes;

const mapStateToProps = (state) => ({
  films: {
    allFilms: getFilms(state),
  },
});

export {AddReview};
export default connect(mapStateToProps)(AddReview);
