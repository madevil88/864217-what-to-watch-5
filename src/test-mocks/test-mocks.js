const reviews = [{
  "id": 1,
  "user": {
    "id": 1,
    "name": `Emely`
  },
  "rating": 4,
  "comment": `This movie is just plain bad. There must be some big payola going round this awards season.`,
  "date": `2020-11-24T16:25:11.950Z`
}, {
  "id": 2,
  "user": {
    "id": 3,
    "name": `Max`
  },
  "rating": 3,
  "comment": `I personally found this movie to be boring.`,
  "date": `2020-11-27T21:02:38.606Z`
}];

const films = {
  allFilms: [{
    "name": `Bronson`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "background_color": `#73B39A`,
    "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    "rating": 3.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "id": 1,
    "is_favorite": false,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }, {
    "name": `Pulp Fiction`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "background_color": `#795433`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "rating": 1.5,
    "scores_count": 1635992,
    "director": `Quentin Tarantino`,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "run_time": 153,
    "genre": `Crime`,
    "released": 1994,
    "id": 2,
    "is_favorite": true,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }],
  filteredFilms: [{
    "name": `Bronson`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "background_color": `#73B39A`,
    "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    "rating": 3.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "id": 1,
    "is_favorite": false,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }, {
    "name": `Pulp Fiction`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "background_color": `#795433`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "rating": 1.5,
    "scores_count": 1635992,
    "director": `Quentin Tarantino`,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "run_time": 153,
    "genre": `Crime`,
    "released": 1994,
    "id": 2,
    "is_favorite": true,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }]
};

export {
  reviews,
  films
};
