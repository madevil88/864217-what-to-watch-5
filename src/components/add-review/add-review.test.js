import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {AddReview} from "./add-review";
import {films} from "../../test-mocks/test-mocks";

const renderer = new ShallowRenderer();

it(`Should AddReview render correctly`, () => {
  renderer.render(
      <AddReview
        films={films}
        id={1}
      />
  );

  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
