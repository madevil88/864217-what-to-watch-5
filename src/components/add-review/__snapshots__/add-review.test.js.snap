// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReview render correctly 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "#73B39A",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <Link
          className="logo__link"
          to="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </Link>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <Link
              className="breadcrumbs__link"
              to="/films/1"
            >
              Bronson
            </Link>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <Memo(Connect(UserBlock)) />
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Bronson"
        height="327"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
        width="218"
      />
    </div>
  </div>
  <WithInputValue
    id={1}
  />
</section>
`;
